cmd/admin-trace.go

```diff
@@ -378,11 +441,45 @@ func matchingOpts(ctx *cli.Context) (opts matchOpts) {
 		responseSize, e = humanize.ParseBytes(ctx.String("filter-size"))
 		fatalIf(probe.NewError(e).Trace(ctx.String("filter-size")), "Unable to parse input bytes.")
 	}
+
+	if ctx.String("rotate-trace-logs-size") != "" {
+		rotateTraceLogsSize, e = humanize.ParseBytes(ctx.String("rotate-trace-logs-size"))
+		fatalIf(probe.NewError(e).Trace(ctx.String("rotate-trace-logs-size")), "Unable to parse input trace logs bytes.")
+	}
+
+	if ctx.String("rotate-trace-logs-frequency") != "" {
+		rotateTraceLogsFrequency, e = strconv.ParseUint(ctx.String("rotate-trace-logs-frequency"), 0, 64)
+		fatalIf(probe.NewError(e).Trace(ctx.String("rotate-trace-logs-frequency")), "Unable to parse input trace logs frequency.")
+	}
+
+	if ctx.String("rotate-trace-logs-directory") != "" {
+		rotateTraceLogsDirectory = ctx.String("rotate-trace-logs-directory")
+	}
+
 	opts.requestSize = requestSize
 	opts.responseSize = responseSize
+	opts.rotateTraceLogsSize = rotateTraceLogsSize
+	opts.rotateTraceLogsFrequency = rotateTraceLogsFrequency
+	opts.rotateTraceLogsDirectory = rotateTraceLogsDirectory
 	return
 }
 
+// Write trace log file
+func writeTracelogFile(log string) {
```

Comment:
It is very inefficient to open+close a file for every write, and I think this will become a bottleneck very quickly.

Same for Stat operation on every write. We keep track of how much we write anyway.

I think keeping the FD open between writes would be reasonable.

Reply:
Thanks for the feedback! Let me know your thoughts on the change please.